### PR TITLE
cloudstack: Add support for ssh_agent_auth

### PR DESCRIPTION
**Pull request by rickard-von-essen**
Requires some more testing.

### CODE REVIEW

**Comment by rickard-von-essen** [comment]
@svanharmelen it would be great if test this on some of your builds.

File: builder/cloudstack/ssh.go [hunk 1]
```diff
@@ -26,32 +28,54 @@ func commHost(state multistep.StateBag) (string, error) {
...
+
+			return &ssh.ClientConfig{
+				User: username,
+				Auth: []ssh.AuthMethod{
+					ssh.PublicKeys(signer),
+				},
+				HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+			}, nil
+
+		} else {
```
Comment (svanharmelen) [inline builder/cloudstack/ssh.go]: You should not use `else` here as the body of this `if` function already returns...

Comment (rickard-von-essen) [inline builder/cloudstack/ssh.go]: Correct!

**Comment by rickard-von-essen** [comment]
I'll ignore the comment because I'm planning to unify all `ssh.go` into one since currently they are unecessarly different. And it's simpler if they are exactly the same and this is just a copy of the one from OpenStack. See #4942

**Comment by svanharmelen** [comment]
@rickard-von-essen so why not fix the linting error in the `OpenStack` builder while your add it, instead of copying the linting error and with that adding more linting errors?

**Comment by rickard-von-essen** [comment]
I'm a simple person and like to focus on one thing at a time. Currently I am happy to focus on making more `ssh.go` look the same to make deduplication easy.

**Pull request merged**